### PR TITLE
Deprecate ember create keys

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
@rwjblue as per request.

Also, unsure if this is labelled correct

**Comment by rwjblue** [comment]
:+1: - And labeled just fine, thank you!@

File: packages/ember-metal/tests/main_test.js [hunk 1]
```diff
@@ -32,3 +32,14 @@ QUnit.test('SEMVER_REGEX properly validates and invalidates version numbers', fu
   validateVersionString('1.11', false);
 });
 
+QUnit.test('Ember.keys is deprecated', function(){
```
Comment (rwjblue) [inline packages/ember-metal/tests/main_test.js]: Missing a space after `function()` here.

File: packages/ember-metal/tests/main_test.js [hunk 2]
```diff
@@ -32,3 +32,14 @@ QUnit.test('SEMVER_REGEX properly validates and invalidates version numbers', fu
...
+  expectDeprecation(function() {
+    Ember.keys({});
+  }, 'Ember.keys is deprecated in-favour of Object.keys');
+});
+
+QUnit.test('Ember.create is deprecated', function(){
```
Comment (rwjblue) [inline packages/ember-metal/tests/main_test.js]: Missing a space after `function()` here (causing JSCS failure).

File: packages/ember-metal/lib/main.js [hunk 3]
```diff
@@ -223,8 +223,8 @@ Ember._Cache = Cache;
 
 Ember.generateGuid    = generateGuid;
 Ember.GUID_KEY        = GUID_KEY;
-Ember.create          = Object.create;
-Ember.keys            = Object.keys;
+Ember.create          = Ember.deprecateFunction('Ember.create is deprecated in-favour of Object.create', Object.create);
```
Comment (rwjblue) [inline packages/ember-metal/lib/main.js]: Should be `Ember.deprecateFunc` I believe (I don't think `Ember.deprecateFunction` is a thing).

File: packages/ember-metal/lib/main.js [hunk 4]
```diff
@@ -223,8 +223,8 @@ Ember._Cache = Cache;
...
+Ember.keys            = Ember.deprecateFunction('Ember.keys is deprecated in-favour of Object.keys', Object.keys);
```
Comment (rwjblue) [inline packages/ember-metal/lib/main.js]: Same as above.

File: packages/ember-metal/tests/main_test.js [hunk 5]
```diff
@@ -32,3 +32,14 @@ QUnit.test('SEMVER_REGEX properly validates and invalidates version numbers', fu
...
+  expectDeprecation(function() {
+    Ember.keys({});
+  }, 'Ember.keys is deprecated in-favour of Object.keys');
+});
+
+QUnit.test('Ember.create is deprecated', function() {
+  expectDeprecation(function() {
+    Ember.keys({});
+  }, 'Ember.create is deprecated in-favour of Object.create');
+})
```
Comment (rwjblue) [inline packages/ember-metal/tests/main_test.js]: Missing semicolon (causing JSHint failure).

File: packages/ember-metal/tests/main_test.js [hunk 6]
```diff
@@ -32,3 +32,14 @@ QUnit.test('SEMVER_REGEX properly validates and invalidates version numbers', fu
...
 
+QUnit.test('Ember.keys is deprecated', function() {
+  expectDeprecation(function() {
+    Ember.keys({});
+  }, 'Ember.keys is deprecated in-favour of Object.keys');
+});
+
+QUnit.test('Ember.create is deprecated', function() {
+  expectDeprecation(function() {
+    Ember.keys({});
```
Comment (rwjblue) [inline packages/ember-metal/tests/main_test.js]: The test says `Ember.create is deprecated` but you are invoking `Ember.keys({})` here (likely copy/paste error).

**Pull request merged**